src/Console/Application.php

```diff
@@ -106,18 +107,39 @@ private function buildDynamicDependencies(InputInterface $input)
                 );
 
                 $infectionConfigFile = $c['locator']->locateAnyOf($configPaths);
-                $dir = \pathinfo($infectionConfigFile, PATHINFO_DIRNAME);
+                $configLocation = \pathinfo($infectionConfigFile, PATHINFO_DIRNAME);
                 $json = file_get_contents($infectionConfigFile);
             } catch (\Exception $e) {
                 $json = '{}';
-                $dir = getcwd();
+                $configLocation = getcwd();
             }
 
-            return new InfectionConfig(json_decode($json), $c['filesystem'], $dir);
+            return new InfectionConfig(json_decode($json), $c['filesystem'], $configLocation);
         };
 
-        $this->container['tmp.dir'] = function (Container $c): string {
-            return $c['tmp.dir.creator']->createAndGet($c['infection.config']->getTmpDir());
+        $this->container['coverage.path'] = function (Container $c) use ($input): string {
+            $existingCoveragePath = $this->getExistingCoveragePath(
+                $input,
+                $c['infection.config']->getPhpUnitConfigDir(),
```

Comment:
Since the --coverage option is a command line option, its location should be relative to where the command is ran from, not the phpunit config location